pkg/util/ha.go

```diff
@@ -0,0 +1,38 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+var (
+	NoOpReplicaChecker = func(ctx context.Context, userID, cluster, replica string, now time.Time) error { return nil }
+)
+
+type ReplicasNotMatchError struct {
```

Comment:
Why don't we just export the existing same error from `distributor` package?

Reply:
Oh, I see we moved it here from there. Why wouldn't we keep it there? I guess it's because you're checking that error in the `handler` now, and we can't depend on `distributor` package because `distributor` probably depends on `util`. That's the evilness of the `util` package.

With the suggestion I made below, you can define the shortcut function in the `distributor` package, move this back where it was before, and avoid any knowledge of `handler` on the internals of our shortcut function (this can also allow us composing shortcuts).